redgenes/gene_annotations.py

```diff
@@ -163,26 +262,72 @@ def annotation_pipeline(df):
         )
 
         ## Step 4: Run barrnap to get {filename}_rrna.gff
-        run_barrnap(unzipped_input, outdir, filename_stem)
+        run_barrnap(unzipped_input, outdir, filename_stem, True)
 
         ## Step 5: Zip fna file
         run_zip_fna(unzipped_input)
 
         ## Step 6: Insert results into tables
         ## prodigal
-        cds_results = extract_gff_contents(f"{Path(outdir) / f'{filename_stem}.gff'}")
+        cds_results = extract_prodigal_results(
+            f"{Path(outdir) / f'{filename_stem}_cds.gff'}"
+        )
+
+        ## kofamscan
+        ko_results = extract_kofamscan_results(
+            f"{Path(outdir) / f'{filename_stem}_kofamscan.tsv'}"
+        )
+
+        ## barrnap
+        rrna_results = extract_barrnap_results(
+            f"{Path(outdir) / f'{filename_stem}_rrna.gff'}"
+        )
 
         ## identifier and md_info
-        sql_identifier = "insert into identifier (filename_full, filepath) values (?, ?) returning genome_id"
+        sql_identifier = "insert into identifier (filename_full, filepath) values (?, ?) returning entity_id"
         args_identifer = [_filename_full, _local_path]
-        sql_md_info = "insert into md_info (genome_id, source, source_detailed, external_accession) values (?, ?, ?, ?)"
+        sql_md_info = "insert into md_info (entity_id, source, source_detailed, external_accession) values (?, ?, ?, ?)"
         args_md_info = [_source, _source_detailed, _external_accession]
 
         with TRN:
             TRN.add(sql_identifier, args_identifer)
-            genome_id = TRN.execute_fetchflatten()
-            args_md_info = genome_id + args_md_info
+            entity_id = TRN.execute_fetchflatten()
+            args_md_info = entity_id + args_md_info
             TRN.add(sql_md_info, args_md_info)
+
+            sql_cds_info = "insert into cds_info (entity_id, contig_id, gene_id, gene_type, start, end, strand, partial, start_type, stop_type, rbs_motif, rbs_spacer, gc_cont, conf, score, cscore, sscore, rscore, uscore, tscore, mscore) values (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)"
+            args_cds_info = [entity_id + res for res in cds_results]
+            TRN.add(sql_cds_info, args_cds_info, many=True)
+
+            colnames_ko_info = [
+                "entity_id",
+                "gene_name",
+                "ko",
+                "threshold",
+                "score",
+                "e_value",
+                "ko_definition",
+            ]
+            sql_ko_info = generate_insert_stmt("ko_info", colnames_ko_info)
```

Comment:
I would encourage having a defined statement. It doesn't seem like dynamic SQL is necessary here, and more generally you have to be very careful about dynamic SQL to avoid injection attacks